compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -11,6 +11,18 @@ class ScalaSettings extends Settings.SettingGroup {
 
   protected def defaultClasspath: String = sys.env.getOrElse("CLASSPATH", ".")
 
+  protected def defaultPageWidth: Int = {
+    val defaultWidth = 80
+    val columnsVar = System.getenv("COLUMNS")
```

Comment:
On Linux at least, this will not be set inside a shell script by default, the best portable way I've found to deal with that is to set it with:
```scala
if command -v tput >/dev/null 2>&1; then
    export COLUMNS="$(tput -Tdumb cols)"
fi
```
in https://github.com/lampepfl/dotty/blob/master/dist/bin/common (I checked and that doesn't impact performance).

Reply:
@smarter I know that [`tput`](https://man7.org/linux/man-pages/man1/tput.1.html) usage but I was hesitant to add it to the Scala 3 launch script. I'm fine if it's ok for you.

I suggest the following change:
```sh
if [ -z ${COLUMNS+x} ] && command -v tput >/dev/null 2>&1; then
    export COLUMNS="$(tput -Tdumb cols)"
fi
```
since it looks like running [`tput`](https://man7.org/linux/man-pages/man1/tput.1.html) ***only once*** per session is enough to activate the automatic update of variable `COLUMNS`.

Can you please confirm the above behavior in your Unix environment ?!
For instance (on your command line, after executing the above command):
1. `$ set COLUMNS=`
2. `$ echo $COLUMNS`  (should print the actual #columms in your terminal)
3. Resize your terminal windows.
4. `$ echo $COLUMNS` (should print the updated #columms in your terminal)

Reply:
Ok, I will add your suggested code for now.

Reply:
Sorry for the delay, looks like I forgot to answer here!

> since it looks like running tput only once per session is enough to activate the automatic update of variable COLUMNS.

I think that the shell itself is responsible for updating $COLUMNS, if I'm in zsh or bash, $COLUMNS get updated automatically even if I reset it, but in dash for example it never gets updated.